include/openssl/digest.h

```diff
@@ -339,7 +346,7 @@ struct env_md_ctx_st {
   // |digest->update|. |digest->update| operates against |md_data| above, but
   // |HMAC_CTX| maintains its own data state in |HMAC_CTX->md_ctx|.
   // |HMAC_Update| also has an additional state transition to handle.
-  void (*update)(EVP_MD_CTX *ctx, const void *data, size_t count);
+  int (*update)(EVP_MD_CTX *ctx, const void *data, size_t count);
```

Comment:
Oh `env_md_ctx_st` is in our public header, do we support users defining there own implementations and this might break them?

Reply:
I don't think the user can implement the internal update function. The update is usually assigned to the EVP_MD md->update, in the case of digest functions https://github.com/aws/aws-lc/blob/82a16b62fa5370686ecd543c787cac442039afec/crypto/fipsmodule/digest/digest.c#L234 and https://github.com/aws/aws-lc/blob/82a16b62fa5370686ecd543c787cac442039afec/crypto/fipsmodule/digest/digest.c#L276
In the case of HMAC it is assigned to the hmac_update; - https://github.com/aws/aws-lc/blob/82a16b62fa5370686ecd543c787cac442039afec/crypto/fipsmodule/evp/digestsign.c#L142